core/src/com/cloud/agent/api/routing/QuaggaConfigCommand.java

```diff
@@ -0,0 +1,86 @@
+//
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+//
+
+package com.cloud.agent.api.routing;
+
+import com.cloud.network.vpc.OSPFZoneConfig;
+import com.cloud.utils.net.cidr.CIDR;
+
+public class QuaggaConfigCommand extends NetworkElementCommand {
+
+    private String routerPublicIp;
+    private String routerName;
+    private String vpnCidr;
+    private CIDR[] tierCidrs;
+    private OSPFZoneConfig zoneConfig;
+
+    public QuaggaConfigCommand(final String routerPublicIp, final String routerName, final String vpnCidr, final CIDR[] tierCidrs, final OSPFZoneConfig qzc) {
```

Comment:
The tier cidrs are provided by the user. I think it would have mode more sense if they were being generated automatically from a given cidr. So a method that breaks a given cidr using a netmask into tiers would be a good application of what you are suggesting.